src/store/cache/useCacheStore.ts

```diff
@@ -221,6 +227,51 @@ const store: StateCreator<CacheStore> = (set, get) => ({
 			set((state) => ({ ...state, loading: { ...state.loading, variables: false } }));
 		}
 	},
+
+	fetchConnections: async (projectId, force) => {
+		const { connections, currentProjectId } = get();
+
+		if (currentProjectId === projectId && !force && connections) {
+			return connections;
+		}
+
+		set((state) => ({
+			...state,
+			currentProjectId: projectId,
+			loading: { ...state.loading, connections: true },
+		}));
+
+		try {
+			const { data: connectionsResponse, error } = await ConnectionService.listByProjectId(projectId!);
+			if (error) {
+				throw error;
+			}
+
+			set((state) => ({
+				...state,
+				connections: connectionsResponse,
+				loading: { ...state.loading, connections: false },
+			}));
+
+			useProjectValidationStore.getState().checkState(projectId!, true);
```

Comment:
Here you already fetched connections, and you'll re-fetch them in the `useProjectValidationStore` again.
You can send the connections from here to the validation store, instead of re-fetching them.

Reply:
I gonna rewrite the cache on the new PR, to avoid redundant API calls